bindings/java/c_kzg_4844_jni.c

```diff
@@ -125,6 +125,39 @@ JNIEXPORT void JNICALL Java_ethereum_ckzg4844_CKZG4844JNI_freeTrustedSetup(JNIEn
   reset_trusted_setup();
 }
 
+JNIEXPORT jbyteArray JNICALL Java_ethereum_ckzg4844_CKZG4844JNI_blobToKzgCommitment(JNIEnv *env, jclass thisCls, jbyteArray blob)
+{
+  if (settings == NULL)
+  {
+    throw_exception(env, TRUSTED_SETUP_NOT_LOADED);
+    return NULL;
+  }
+
+  size_t blob_size = (size_t)(*env)->GetArrayLength(env, blob);
+  if (blob_size != BYTES_PER_BLOB)
+  {
+    throw_invalid_size_exception(env, "Invalid blob size.", blob_size, BYTES_PER_BLOB);
+    return NULL;
+  }
```

Comment:
Hmm, why do we need to do this verbose checking for basic invariants? My suggestion would be to replace these 11 lines of code with two simple assertions:
```suggestion
assert(settings);
assert((size_t)(*env)->GetArrayLength(env, blob) == BYTES_PER_BLOB)
```

I don't understand why we need to fail gracefully if the caller has not initialized the trusted setup, or has provided us with junk data. Let's ensure we don't segfault (i.e. assert) but no reason to do graceful handling of these bad cases.

Same goes for similar boilerplate in all the other functions. I think this change will result in a non-trivial simplfication of this file.